osu.Game.Rulesets.Mania/Replays/ManiaAutoGenerator.cs

```diff
@@ -77,10 +78,34 @@ public override Replay Generate()
 
         private IEnumerable<IActionPoint> generateActionPoints()
         {
-            foreach (var obj in Beatmap.HitObjects)
+            for (int i = 0; i < Beatmap.HitObjects.Count; i++)
             {
-                yield return new HitPoint { Time = obj.StartTime, Column = obj.Column };
-                yield return new ReleasePoint { Time = ((obj as IHasEndTime)?.EndTime ?? obj.StartTime) + RELEASE_DELAY, Column = obj.Column };
+                var currentObject = Beatmap.HitObjects[i];
+                var nextObjectInTheSameColumn = getNextObjectInTheSameColumn(i);
```

Comment:
too long name, also can be simplified with linq
```suggestion
                var nextObjectInColumn = Beatmap.HitObjects.SkipWhile(h => h != currentObject).Skip(1).FirstOrDefault(h => h.Column == currentObject.Column);
```

Reply:
Using linq will make this O(n^2), and allocate quite a ton. Definitely prefer the way it was implemented instead.